### PR TITLE
Fix IllegalStateException: for NotificationRemovalService

### PR DESCRIPTION
**Pull request by andrews-moc**
[MOB-2837](https://glia.atlassian.net/browse/MOB-2837)

**Crash details:**
```
java.lang.IllegalStateException: Not allowed to start service Intent { com.glia.widgets.core.notification.NotificationRemovalService }: app is in background
```
Example from Sentry is [here](https://glia.sentry.io/issues/4766376086/?alert_rule_id=967509&alert_timestamp=1703689998756&alert_type=email&environment=prod-us&notification_uuid=097e6631-f490-4032-8f8f-622acb0a80d3&project=1887832&referrer=alert_email).

This happens if an app is in the background for 60+ seconds and NotificationRemovalService is attempted to be started.

**What was solved?**
I came to conclusion that there is no need in `NotificationRemovalService` at all, since some time ago we have added the `notificationManager.cancel(NotificationFactory.CALL_NOTIFICATION_ID)` call before starting the service. Hopefully, I am not missing anything. Tested various scenarios - call notification is being dismissed when call is ended.

**Release notes:**

 - [ ] Feature
 - [ ] Ignore
 - [x] Release notes (Is it clear from the description here?)
 - [ ] Migration guide (If changes are needed for integrator already using the SDK - what needs to be communicated? Add underneath please)

**Additional info:**

 - [ ] Tests updated? Added? Unit, acceptance, snapshots?
 - [ ] Logging for future troubleshooting of client issues added?

**Release note:**
Fix IllegalStateException: for NotificationRemovalService

[MOB-2837]: https://glia.atlassian.net/browse/MOB-2837?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by DavDo** [comment]
Have you tried to kill the app during the engagement?

File: widgetssdk/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -87,8 +87,6 @@
             android:name=".core.screensharing.MediaProjectionService"
             android:foregroundServiceType="mediaProjection" />
 
-        <service android:name=".core.notification.NotificationRemovalService" />
```
Comment (AndriiHorishniiMOC) [inline widgetssdk/src/main/AndroidManifest.xml:90]: No service - no issue :)

**Review by AndriiHorishniiMOC — APPROVED** [review]
I like the way it fixed 👍

**Comment by andrews-moc** [comment]
> Have you tried to kill the app during the engagement?

@DavDo Indeed, this is the case which we need the service for. Notification is not removed if app is killed. So this approach is not appropriate. Thank you.

**Pull request closed without merging**